osdep/compiler.h

```diff
@@ -23,10 +25,4 @@
 #define PRINTF_ATTRIBUTE(a1, a2) __attribute__ ((format (gnu_printf, a1, a2)))
 #endif
 
-#ifdef __GNUC__
-#define MP_ASSERT_UNREACHABLE() (assert(!"unreachable"), __builtin_unreachable())
-#else
-#define MP_ASSERT_UNREACHABLE() (assert(!"unreachable"), abort())
-#endif
```

Comment:
clang also defines `__GNUC__` so it's not really "fixing" anything. Just moving it around.

Reply:
> clang also defines `__GNUC__` so it's not really "fixing" anything. Just moving it around.

clang does not define `__GNUC__` if it's not targeting GNU target. There is nothing GNU about `x86_64-pc-windows-msvc` target for example.

Reply:
I've added note about it in commit message.